### PR TITLE
Add function to specify build-tool generator

### PR DESCRIPTION
**Pull request by caipre**
Required for https://github.com/rust-lang/rust/issues/32809

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -237,7 +245,9 @@ impl Config {
                 // On MinGW we need to coerce cmake to not generate a visual
                 // studio build system but instead use makefiles that MinGW can
                 // use to build.
-                cmd.arg("-G").arg("MSYS Makefiles");
+                if self.generator.is_none() {
+                    self.generator("MSYS Makefiles");
```
Comment (alexcrichton) [inline src/lib.rs]: Could this actually just pass down `-G` directly? In theory we don't want to do *too* much mutation as part of this call of the `Config` structure itself

File: src/lib.rs [hunk 2]
```diff
@@ -262,7 +272,12 @@ impl Config {
         } else if msvc {
             // If we're on MSVC we need to be sure to use the right generator or
             // otherwise we won't get 32/64 bit correct automatically.
-            cmd.arg("-G").arg(self.visual_studio_generator(&target));
+            if self.generator.is_none() {
+                self.generator(Config::visual_studio_generator(&target));
+            }
+        }
+        if self.generator.is_some() {
```
Comment (alexcrichton) [inline src/lib.rs]: This could be simplified with:

```
if let Some(ref generator) = self.generator {
    cmd.arg("-G").arg(generator);
}
```

**Comment by alexcrichton** [comment]
Just a few minor nits but otherwise looks good to me! Could you also back out the version number change for now? I like to do separate standalone commits with that

**Comment by caipre** [comment]
Thanks for the review, I've pushed a new commit.

**Pull request merged**